cp-main.go

```diff
@@ -164,6 +158,21 @@ func doCopy(cpURLs copyURLs, progressReader *progressBar, accountingReader *acco
 		// Set up progress reader.
 		progress = progressReader.ProgressBar
 	}
+	// If source size is <= 5GB use server side copy/CopyObject API for s3.
+	if length <= 5*1024*1024*1024 {
```

Comment:
Save this a 'const' at the top - additionally you need to verify if the source and target are pointing to same aliased URL. Otherwise copying in S3 to GCS (Google Cloud Storage) or S3 to Minio Server would fail.